backend/src/email/models/email-message-transactional.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  Column,
+  DataType,
+  ForeignKey,
+  Model,
+  Table,
+} from 'sequelize-typescript'
+import { User } from '@core/models'
+
+export enum TransactionalEmailMessageStatus {
+  Unsent = 'UNSENT',
+  Accepted = 'ACCEPTED',
+  Sent = 'SENT',
+  Bounced = 'BOUNCED',
+  Delivered = 'DELIVERED',
+  Opened = 'OPENED',
+  Complaint = 'COMPLAINT',
+}
+
+export type AttachmentsMetadata = AttachmentMetadata[]
+
+interface AttachmentMetadata {
+  fileName: string
+  fileSize: number // in bytes
+  hash: string
+}
+
+@Table({
+  tableName: 'email_messages_transactional',
+  underscored: true,
+  timestamps: true,
+})
+export class EmailMessageTransactional extends Model<EmailMessageTransactional> {
+  @Column({
+    type: DataType.BIGINT,
+    autoIncrement: true,
+    primaryKey: true,
+    allowNull: false,
+  })
+  id: number
+
+  @ForeignKey(() => User)
+  @Column({ type: DataType.STRING, allowNull: false })
+  userEmail: string
+
+  @Column({ type: DataType.STRING, allowNull: false })
+  fromName: string
+
+  @Column({ type: DataType.STRING, allowNull: false })
+  fromAddress: string
+
+  @Column({ type: DataType.STRING, allowNull: false })
+  recipient: string
+
+  @Column({ type: DataType.JSON, allowNull: false })
+  params: Record<string, string>
+
+  @Column({ type: DataType.STRING, allowNull: true })
+  messageId: string | null
+
+  // null means this column has yet to be processed
+  // empty array means there are no attachments
```

Comment:
This is unnecessary as it's very unlikely that this column will be null under this circumstance, and if it does - we have a bug to fix rather than an audit need. It's not worth it to require another db update as we have application logs for this kinda things